schema/gen/go/genList.go

```diff
@@ -186,11 +186,11 @@ func (g listGenerator) EmitNodeMethodListIterator(w io.Writer) {
 			idx  int
 		}
 
-		func (itr *_{{ .Type | TypeSymbol }}__ListItr) Next() (idx int, v ipld.Node, _ error) {
+		func (itr *_{{ .Type | TypeSymbol }}__ListItr) Next() (idx int64, v ipld.Node, _ error) {
 			if itr.idx >= len(itr.n.x) {
 				return -1, nil, ipld.ErrIteratorOverread{}
 			}
-			idx = itr.idx
+			idx = int64(itr.idx)
```

Comment:
Yeah, these upcasts being needed in iterators all the time are a bit weird.  You explained it well already in the PR description though, and the logic seems sound.  :+1: